datafusion/physical-expr/src/aggregate/count_distinct.rs

```diff
@@ -206,7 +206,10 @@ impl Accumulator for DistinctCountAccumulator {
             )
         });
 
-        Ok(cols_out)
+        Ok(cols_out
+            .iter()
+            .map(|v| AggregateState::Scalar(v.clone()))
+            .collect())
```

Comment:
You can probably avoid a copy like:

```suggestion
        Ok(cols_out
            .into_iter()
            .map(|v| AggregateState::Scalar(v))
            .collect())
```